projects/igniteui-angular/src/lib/grids/columns/column.component.ts

```diff
@@ -1682,14 +1681,19 @@ export class IgxColumnComponent implements AfterContentInit, OnDestroy {
             return false;
         }
 
-        this._pinned = true;
-        this.pinnedChange.emit(this._pinned);
         this._unpinnedIndex = grid._unpinnedColumns.indexOf(this);
         const rootPinnedCols = grid._pinnedColumns.filter((c) => c.level === 0);
         index = index !== undefined ? index : rootPinnedCols.length;
-        const targetColumn = grid._pinnedColumns[index];
-        const args = { column: this, insertAtIndex: index, isPinned: true };
-        grid.onColumnPinning.emit(args);
+        const args: IPinColumnCancellableEventArgs = { column: this, insertAtIndex: index, isPinned: true, cancel: false };
```

Comment:
The column is not pinned yet, so we should change isPinned to false here.

Reply:
`isPinned` as part of the event arguments is meant to indicate the new value that will be applied in the end. I know the name can be confusing, but I kept it in order to keep consistency between `IPinColumnEventArgs` and the new `IPinColumnCancellableEventArgs`.

Reply:
It is more subjective the thing we are discussing. Again IMO ing events shows the state before it is applied. Because If I decide to cancel it I will have in the arguments the value saying that the column is pinned which is not true. 

I may be biased a little bit, but that is why in the crud service we are exposing the old or current value and the new one, so the user is not getting confused

Reply:
But if we are seeing it from the perspective where we are saying that this is the newValue really, it may be the right implementation.